### ISSUE TITLE
Week 16 sprint

### ISSUE THREAD

**Issue by sofer** [foundersandcoders/london-curriculum#15]
There are [quite a few potential partners](https://github.com/foundersandcoders/london-employers/labels/Graduate%20Project%3A%20in%20discussion). Please do let me know about any more that you know of.

**Comment by sofer** [comment]
@nelsonic has offered this as a dwyl week with the guarantee of at least one graduate project. I am very keen for us take dwyl up on this offer. More details to follow.

**Comment by nelsonic** [comment]
@sofer does @dwyl quality as an Employer for the list you've linked to above?

**Comment by iteles** [comment]
Note for reference: this is the week of 5th June 2017.

**Comment by nelsonic** [comment]
As discussed with @sofer & @iteles the plan is to organise Week 16 into something _unforgettable_!
I have a few questions we need to answer before proceeding with preparations. 
The _principal_ question is:

# What is the `main` Objective for Week 16?

### Further Clarifying Questions

1. Is it _correct_ that "Week 16" is  the "_last week_" the cohort is together before "graduation"?
2. Does `each` person in the cohort have a ["**Skills Summary**"](https://github.com/SimonLab/CV#skills) in their (_GitHub-based_) **CV** similar to how @Neats29, @heron2014 & @SimonLab have done (_many people from F&C 4,5,6 have this style of CV, but it does not appear to be the norm in F&C 10_)
3. What is `each` person's **objective _after_** "Week 16"? (_do we have a **record** of this somewhere_?)

**Comment by bradreeder** [comment]
@nelsonic 

1. That's right. The two weeks after that are a break before FAC11 starts. But some students could be working on graduate projects then depending on start date. 👍 
2. I don't think so, but can recommend it to them.
3. I should be able to get back to you on this this week. We're having our third pastoral care round and the student's plans for post week-16 is one of the main subjects we need to cover. I imagine most will either be wanting to build a portfolio by working on their first graduate projects, or looking for their first jobs.

I will try to catch up with you some time in the coming week. 👍

**Comment by nelsonic** [comment]
We _briefly_ discussed the possibility of giving people a list "_pre-learning_" for the week.
I don't want to list anything _yet_ until we clarify what `each` person will be doing `after` F&C.
I _also_ think it would be _useful_ to re-watch people's application videos prior to the week. 
@iteles let me know if/when you have time for this. 👍

**Comment by nelsonic** [comment]
Pre Week 16 Learning Material
+ [ ] https://github.com/dwyl/learn-elm-architecture-in-javascript
+ [ ] https://github.com/dwyl/learn-tachyons
+ [ ] https://github.com/dwyl/learn-elm ("_Bonus Level!_")

**Comment by sofer** [comment]
@nelsonic this looks totally amazing! (very sad that I might be in Nazareth/Gaza for it)

**Comment by sofer** [comment]
Quick question. Have those learning resources been posted into the FAC10 gitter channel yet? There's a fair amount of material there and it would be good to make sure people have time to absorb it.
 @bradreeder @nelsonic

**Comment by bradreeder** [comment]
@sofer I've just done it. The students will have a few hours free on Tuesday morning so I'm going to recommend they get started on the materials then.

**Comment by bradreeder** [comment]
@nelsonic An external speaker has asked if they can come in to pitch a project to the students at 5pm on either Monday or Tuesday of next week. Would that be okay and which day would be better for you? : )

I've given the students the materials you recommended and asked them to get started on it tomorrow morning. 👍

**Comment by bradreeder** [comment]
@nelsonic An external speaker has asked if they can come in to pitch a project to the students at 5pm on either Monday or Tuesday of next week. Would that be okay and which day would be better for you? : )

I've given the students the materials you recommended and asked them to get started on it tomorrow morning. 👍

**Comment by nelsonic** [comment]
Hi @foundersandcoders/fac10 Team(s)! The day has _finally_ arrived! 🎉 
You've accumulated 15 weeks of knowledge/experience (_plus all that pre-course prep_!) 
and now you `make` something useful/practical with your skills! 

This week the challenge is to build something that "_solves a **real challenge**_" 
for yourselves (_or a loved one_) using a _real_ API: https://github.com/dwyl/api 😮 
The API is a RealTime "General Purpose" Data Store which means you can store _any_ type of data!

## _Why_? 

Building web/mobile applications is the "bread and butter" for _most_ developers.
The _users_ of your apps will not give you _any_ credit for designing a "_beautiful API_" 
if the UX is bad, so this week you will 

### _What's in it For Me (the F&C Student)...?_

+ A chance to work with _experienced_ developers who can answer your questions in minutes instead of you having to struggle for hours/days! 
+ A portfolio piece you can show future team mates, prospective employers or co-founders.
+ The potential to "_build out_" the feature into a complete app (_paid work for @dwyl_) 

## _What_?

Build a web-based front-end mobile-first application that solves a specific real-world problem.

> If you don't already have a problem in mind, speak to Nelson about tackling one of the 4 @dwyl challenges. 

## _Who_?

+ F&C Cohort 10 members
+ dwyl team members will answer any questions regarding the tech and API as required (_please open issues on GitHub when you have questions_) 

## _How_?

1. Identify the Problem/Challenge you want to solve (25 mins)
(_If you don't already have a problem, ask Nelson for one of the dwyl challenges_)
2. Describe the problem in a single user-focussed sentence on a Post-it
3. Share your problem statement with @dwyl team members. (_by posting a comment in this issue_)
(_note: it has to be a real problem/challenge; no "dating" apps_) 
4. Find your team members (min 2 max 4)
5. Start your design sprint! 
6. On day 3 start integrating your app with the API
7. Friday: demo your app! 


### API?

You will not _need_ to use the API for the first 2 days of this week as you will be building your app(s) on your local machines exclusively front-end UX/UI.

We (_@dwyl team_) will give you a _demo_ of the API on Wednesday Morning (_7th June_)
and show you how to use both the Dev and Prod versions of the API. 

If you get _curious/bored_ in the next 48h, read: https://github.com/dwyl/learn-api-design

<br /><br />

### _Very_ Nice-to-Have (_AKA "Bonus Level Criteria_")

These are not _requirements_ but if you achieve them (_or show progress towards them_) you will _unlock_ the _next level_. 

1. Use `Elm`
2. Offline-first (_e.g using Progressive Web Apps techniques_)
3. Automatically save data to the API each time the user interacts with the app.
4. Allow the same user to use two (_or more devices_) simultaneously and use the WebSocket API to sync the data. 
5. Create a Pull Request adding your knowledge to _any_ of the dwyl `learn-...` resources.

Anyone that meets all criteria will _automatically_ meet @dwyl Apprentice requirements
(_i.e. "fast track" to becoming an experienced - and well-paid - developer!_)
This will include a _paid_ trip to "dwyl summer" in Portugal to `continue` working on you App/Project (_or the dwyl app_) see: https://github.com/dwyl/summer/issues/49#issuecomment-281198810 for a _glimpse_ of the best week(s) of your 2017!

**Comment by ZooeyMiller** [comment]
User story : 
>"As a person with variable mood I want a way to record my mood at intervals of my choosing, so that I can view a representation of my mood over time at a later date".

More specifics: 
* Must be __fast__ and __easy__ so that it never feels like a chore.

**Comment by edgarfinn** [comment]
Music Recognition for audio files and / or streams using the [ACRCloud API](https://www.acrcloud.com/), the [Gracenote API](https://developer.gracenote.com/) or the [Houndify API](https://www.houndify.com/)

User story:
> "As a music fanatic and "crate digger" always in search of new music, I want to be told the songs / tracks that are used in my favourite podcasts, so that I can discover more great music similar to my taste"

**Comment by pbywater** [comment]
Portfolio app for ipad with a way to manage meetings and record quantitative and qualitative data on the interactions the potential client has had with the images. 

User story:
> As a freelancer with a visual portfolio, I currently don't have a way to track analytics on how potential clients interact with my portfolio during meetings. I could use this to keep track of leads and optimise my portfolio.

@PiotrBerebecki @Akin909 @lucyrose93

**Comment by mattlub** [comment]
Problem:
> As a group trying to assess the opinion of the members on a certain question, I want an easy way to create a poll and see the results.

Solution:
> A chat app with seamlessy integrated support for polls among members of the chat.

**Comment by nelsonic** [comment]
@mattlub the polling feature is _really_ useful and solves a real problem. ✅ 
I would _suggest_ that you consider making it a "_standalone_" feature that _can_ _optionally_ be integrated into a chat app or shared in other places the same way a twitter poll can be:
e.g: https://github.com/dwyl/learn-elm/issues/10

As a feature it's _generally_ applicable to _many_ situations/contexts.
The question you have to ask is: 
+ Will voting be based access to a specific URL?
+ Will you control who is allowed to vote based on authentication?
+ How will you ensure people do not vote multiple times? (_do you want the API to "handle" this?_)
+ What does the UX/UI need to have to:
  + CREATE the Poll with specific question and set of answers and optionally set a deadline.
  + Share it with the relevant people using a short link.
  + Allow people to Vote
  + Display the results once they have voted and once the poll ends/closes.

If you focus on the UX/UI for the next 2 days and let us know what `data` you want to store on Wednesday morning we can _ensure_ that the API is capable of storing/retrieving the data to support your feature. Sound good?

**Comment by mattlub** [comment]
@nelsonic agreed a standalone is arguably more useful, but the wrt the questions:
>Will voting be based access to a specific URL?
Will you control who is allowed to vote based on authentication?
How will you ensure people do not vote multiple times? (do you want the API to "handle" this?)

They would all be handled by the auth of a certain chat room.. I guess the ideas are fundamentally quite different. Thoughts?

**Comment by alexis-l8** [comment]
User Story:

> - As a user, I want to be able to post a task which contains a list of more specific instructions for this task to be completed. 
> - I also want to be able to see the list checked off in real time. 

> - As a user, I want to see all available tasks in London and be able to filter the by geography or renumeration. 
> - I want to be able to check individual items off the list as I do them

Stretch goals:
> - Elm
> - Ability for payment (bitcoin/ ether)
> - Real-time web sockets chat

Team: 

@Samatar26 🔥 🔥 🔥

**Comment by yvonne-liu** [comment]
A contraceptive information app that’s easy to use with options, includes a calendar/diary tracking option to better understand your cycles, and make informed sexual health decisions: 

>As a girl/woman who is considering using/changing contraceptives, I want to know all of the options available to me so that I can make an informed decision about my sexual reproductive health. 
 
>As a girl/woman I want to be able to track the effects of contraceptives on my physical and mental health so that I can make a decision as to whether my current form of contraceptives is helping me. 

@bo-bok @smarthutza @joeylouise

**Comment by nelsonic** [comment]
@foundersandcoders/fac10 / @bradreeder can someone _confirm_ that everyone in the cohort has a project/team for the week?

@mattlub to answer your question: it _depends_ if you _want_ to make the Auth for Polling the same as Auth for Chat that's fine. i.e if a person has access to a given chat "room" then they automatically have access to vote on the Poll. I would _suggest_ that this is a feature that gets added _later_ once the core Poll UI/UX is done. Ask yourself the question: what is the "risk" or "downside" of someone voting on a poll during MVP?

**Comment by Samatar26** [comment]
@nelsonic can confirm everybody has a team and project for the week :+1:

**Comment by mattlub** [comment]
@nelsonic we are going for the standalone poll- there is no "risk" of someone voting on a poll during MVP in the intended use cases 😄

**Comment by Samatar26** [comment]
@nelsonic can we have a demo on the Dwyl api?

**Comment by ZooeyMiller** [comment]
Does anyone know the plan for tomorrow in terms of when presentations and stuff might be? 
Just so we know what time of the day we are working up until and stuff.

**Comment by nelsonic** [comment]
@ZooeyMiller let's aim to start presenting at 16:00 🕓

**Comment by yvonne-liu** [comment]
@nelsonic Just wanted to let you know we'll have an external speaker at 17:30, but can be pushed back if necessary. Chill o' 🕦 .

**Comment by nelsonic** [comment]
@yvonne-liu are you referring to having an external speaker tomorrow?

**Comment by oliverjam** [comment]
@nelsonic That's right

**Comment by nelsonic** [comment]
@Samatar26 

Visit: https://dwylapi.herokuapp.com/rows

Create a new Person (User)
```
curl -H "Content-Type: application/json" -X POST -d '{"person": {"name":"Simon","email":"simon@dwyl.io"}}' https://dwylapi-noauth.herokuapp.com/people
```

Now create a Row (record):
```
curl -H "Content-Type: application/json" -X POST -d '{"rows": {"id":1, "title":"Simon Says Hi!","body":"This is my amazing post!", "people_id":"1"}}' https://dwylapi-noauth.herokuapp.com/rows
```
You should see your record when you visit:
https://dwylapi-noauth.herokuapp.com/rows

**Comment by yvonne-liu** [comment]
@nelsonic Are we starting presentation at 16:00? ✨

**Comment by nelsonic** [comment]
@yvonne-liu who's going first? 🥇 😉

**Comment by nelsonic** [comment]
A couple of people have _already_ shown their interest in the @dwyl **Apprenticeship**.
If anyone `else` is interested please comment on: https://github.com/dwyl/hq/issues/356

**Comment by jsms90** [comment]
This issue was moved to foundersandcoders/london-programme#364
